com.unity.multiplayer.mlapi/Tests/Editor/IndexAllocatorTests.cs

```diff
@@ -0,0 +1,109 @@
+using NUnit.Framework;
+using UnityEngine;
+
+namespace MLAPI.EditorTests
+{
+    public class FixedAllocatorTest
+    {
+        [Test]
+        public void SimpleTest()
+        {
+            int pos;
+
+            var allocator = new IndexAllocator(20000, 200);
+            allocator.DebugDisplay();
+
+            // allocate 20 bytes
+            Assert.IsTrue(allocator.Allocate(0, 20, out pos));
+            allocator.DebugDisplay();
+            Assert.IsTrue(allocator.Verify());
+
+            // can't ask for the same index twice
+            Assert.IsFalse(allocator.Allocate(0, 20, out pos));
+            Assert.IsTrue(allocator.Verify());
+
+            // allocate another 20 bytes
+            Assert.IsTrue(allocator.Allocate(1, 20, out pos));
+            allocator.DebugDisplay();
+            Assert.IsTrue(allocator.Verify());
+
+            // allocate a third 20 bytes
+            Assert.IsTrue(allocator.Allocate(2, 20, out pos));
+            allocator.DebugDisplay();
+            Assert.IsTrue(allocator.Verify());
+
+            // deallocate 0
+            Assert.IsTrue(allocator.Deallocate(0));
+            allocator.DebugDisplay();
+            Assert.IsTrue(allocator.Verify());
+
+            // deallocate 1
+            allocator.Deallocate(1);
+            allocator.DebugDisplay();
+            Assert.IsTrue(allocator.Verify());
+
+            // deallocate 2
+            allocator.Deallocate(2);
+            allocator.DebugDisplay();
+            Assert.IsTrue(allocator.Verify());
+
+            // allocate 50 bytes
+            Assert.IsTrue(allocator.Allocate(0, 50, out pos));
+            allocator.DebugDisplay();
+            Assert.IsTrue(allocator.Verify());
+
+            // allocate another 50 bytes
+            Assert.IsTrue(allocator.Allocate(1, 50, out pos));
+            allocator.DebugDisplay();
+            Assert.IsTrue(allocator.Verify());
+
+            // allocate a third 50 bytes
+            Assert.IsTrue(allocator.Allocate(2, 50, out pos));
+            allocator.DebugDisplay();
+            Assert.IsTrue(allocator.Verify());
+
+            // deallocate 1, a block in the middle this time
+            allocator.Deallocate(1);
+            allocator.DebugDisplay();
+            Assert.IsTrue(allocator.Verify());
+
+            // allocate a smaller one in its place
+            allocator.Allocate(1, 25, out pos);
+            allocator.DebugDisplay();
+            Assert.IsTrue(allocator.Verify());
+        }
```

Comment:
I would try and test limits. What happens when you allocate somewhere invalid? What happens when you allocate just at the limit?
Other invalid cases? negative values?

Reply:
negative value was a missing case, and the API was not checking for it. I'll fix that. Thank you!